packages/components/src/utils/getTableOfContents.ts

```diff
@@ -8,21 +8,36 @@ export const getTableOfContents = (
   content: IsomerComponent[],
 ): HeadingLink[] => {
   return content.flatMap((block) => {
-    if (block.type !== "prose" || !block.content) {
-      return []
+    if (
+      (block.type === "infocards" ||
+        block.type === "infocols" ||
+        block.type === "infopic" ||
+        block.type === "keystatistics") &&
+      block.title
+    ) {
+      return [
+        {
+          content: block.title,
+          anchorLink: `#${block.id}`,
+        },
+      ]
     }
 
-    const result = []
+    if (block.type === "prose" && block.content) {
```

Comment:
should we still have a defensive check here? previously we allowed **only** `prose` with `content` - right now, we allow 4 blocks (assuming they have title), then we make the assumption that **the remaining blocks are prose with content**. 

what happens if we add a new block?

Reply:
Hmm we are only selectively adding support for certain blocks, not all blocks will be eligible to be part of the table of contents. If there is a new block, it will fallback to the last return which is the empty array `[]`.